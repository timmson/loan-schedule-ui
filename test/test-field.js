import React from "react"
import renderer from "react-test-renderer"
import Field from "../src/field"

describe("Field should", () => {

	let component = null

	beforeAll(() => {
		component = renderer.create(
			<Field
				name={"name"}
				description={"value"}
				value={"value"}
				onChange={(event) => event}
			>X</Field>
		)
	})

	test("trigger change", () => {
		const  expected = 1

		const result = component.root.findByType("input").props.onChange(expected)

		expect(result).toEqual(expected)
	})

	test("equal to snapshot", () => {
		expect(component.toJSON()).toMatchSnapshot()
	})

	afterAll(() => component.unmount())

})