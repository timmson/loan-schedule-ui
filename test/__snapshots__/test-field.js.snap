// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Field should equal to snapshot 1`] = `
Array [
  <div
    className="col-sm-3 pt-2"
  >
    <label
      htmlFor="name"
    >
      value
    </label>
  </div>,
  <div
    className="col-sm-3"
  >
    <input
      className="form-control"
      name="name"
      onChange={[Function]}
      value="value"
    />
  </div>,
]
`;
